### PR TITLE
tap-info: handle APFS returning hash order

### LINKED ISSUE #3311: Homebrew sometimes does not sort alphabetically when expected to
- [x] Confirmed this is a problem with running a `brew` command and not `brew install`ing or the post-install behaviour of one or more formulae? If it's a formulae-specific problem please file this issue at the relevant tap e.g. for Homebrew/homebrew-core https://github.com/Homebrew/homebrew-core/issues/new
- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed all issues and retried your prior step?
- [x] Ran `brew config` and `brew doctor` and included their output with your issue?

shortened for brevity:
```
⇒  brew config
HOMEBREW_VERSION: 1.3.5
ORIGIN: https://github.com/Homebrew/brew
HEAD: 91ab116ace7f4f97d3440190463c93be9ec6d675
Last commit: 5 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 8b40d2317edc730da739b66368d493812fd63ac5
Core tap last commit: 3 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/Homebrew
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: octa-core 64-bit haswell
Homebrew Ruby: 2.3.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/bin/ruby
Clang: 9.0 build 900
Git: 2.14.2 => /usr/local/bin/git
Perl: /usr/bin/perl
Python: /usr/local/opt/python/libexec/bin/python => /usr/local/Cellar/python/2.7.14/Frameworks/Python.framework/Versions/2.7/bin/python2.7
Ruby: /usr/bin/ruby => /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/bin/ruby
Java: 1.8.0_144
macOS: 10.13-x86_64
Xcode: 9.0
CLT: N/A
X11: N/A

⇒  brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry and just ignore them. Thanks!

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libaws*.dylib

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/python3.6m/greenlet/greenlet.h
  /usr/local/include/python2.7/greenlet/greenlet.h
  /usr/local/include/aws/*

Warning: Unbrewed .pc files were found in /usr/local/lib/pkgconfig.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected .pc files:
  /usr/local/lib/pkgconfig/aws*.pc
```

After updating to macOS High Sierra, `brew search` with no search terms does not list available formulae alphabetically. the paths in `brew doctor` and `brew cleanup`, similarly are not sorted alphabetically. Other commands  This is likely due to the shift from HFS+ to APFS.

### PR DESCRIPTION
**Pull request by jmsundar**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Fixes #3311.

### CODE REVIEW

File: Library/Homebrew/cmd/tap-info.rb [hunk 1]
```diff
@@ -31,9 +30,9 @@ def tap_info
     end
 
     if ARGV.json == "v1"
-      print_tap_json(taps)
+      print_tap_json(taps.sort_by {|tap| tap.to_s})
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/tap-info.rb]: Run `brew style` and it'll point out a couple of style tweaks for this line (and the change below, too).

Comment (jmsundar) [reply Library/Homebrew/cmd/tap-info.rb]: Ah I see, thanks.

**Comment by MikeMcQuaid** [comment]
Thanks for your first contribution to Homebrew, @jmsundar! Without people like you submitting PRs we couldn't run this project. You rock!

**Comment by DomT4** [comment]
Thanks for grabbing this last one @jmsundar. Life has not been consistently kind to my schedule lately 😓.

**Pull request merged**